### ISSUE TITLE
Bug after update

### ISSUE THREAD

**Issue by Tongzhoutz** [lervag/vimtex#1864]
I am running into a problem after updating my init.vim. The weird thing is that the tex file can be successfully compiled, but several error messages popped up at the same time.

Now what I can investigate is that this error may be related to vim-snippets, because the designated snippets in vim-snippets cannot be invoked in tex files. 

The attached are error messages. So I was wondering if anyone can shed lights on the issue?

<img width="1971" alt="Screen Shot 2020-11-24 at 14 08 42" src="https://user-images.githubusercontent.com/61123232/100140191-9cf46d00-2e5e-11eb-8037-cc035e6ce26b.png">

**Comment by lervag** [comment]
> I am running into a problem after updating my init.vim. ...

You did not read [the issue template](https://github.com/lervag/vimtex/blob/master/.github/ISSUE_TEMPLATE/issue_report.md)...

In particular, you've missed this important point:

> Provide relevant files and commands in detail, so everybody can reproduce the issue! The following are examples of minimal input files. To use the minimal vimrc files: ...

**Comment by funnydman** [comment]
@lervag I've got the same issue as above after updating the plugin

**Comment by lervag** [comment]
> @lervag I've got the same issue as above after updating the plugin

I'm sorry about the inconvenience, but it seems you also did not read my last comment. I'm happy to help, but in order to help I **need** more information. In particular, I need the relevant configuration and steps required to reproduce.

**Comment by danielfalbo** [comment]
Having the same issue and it's being really hard to reproduce because does not happen all the time. However seems like this way to reproduce is pretty stable on my machine and I hope you'll be able to recreate the error too.

1. Open neovim with a new file, foo.tex
```
nvim foo.tex
```
2. Paste the following in neovim
```tex
\RequirePackage[l2tabu, orthodox]{nag}
\documentclass[a4paper]{article}

\usepackage[T1]{fontenc}
\usepackage[shortlabels]{enumitem}
\usepackage{lmodern}
\usepackage{textcomp}
\usepackage{amsmath, amssymb}
\usepackage{graphicx}
\usepackage{geometry}
\usepackage{microtype}
\usepackage{booktabs}
\usepackage{xcolor}
\usepackage{array}
\usepackage{listings}
\usepackage{amsthm}
\usepackage{epigraph}
\usepackage{tikz}
\usepackage{fontawesome}

\usepackage{import}
\usepackage{xifthen}
\pdfminorversion=7
\usepackage{pdfpages}
\usepackage{transparent}
\newcommand{\incfig}[1]{%
    \def\svgwidth{\columnwidth}
    \import{./figures/}{#1.pdf_tex}
}

\pdfsuppresswarningpagegroup=1

\title{\Huge Hello, world!}
\author{\Large Daniel Falbo --- \texttt{danielfalbo@engineer.com}}
\date{\the\year{}}

\begin{document}
\maketitle
\end{document}
```
3. Hit `:w`
4. Hit `<localleader>ll`
5. Get the error (wait for it, it may initially say "Compilation completed" but after a little bit the error should occur)

It's really weird and it does not happen if I do not include the packages and the maketitle in the file.

**Comment by danielfalbo** [comment]
I believe the error has been introduced by the neovim update, not by the vimtex update.

**Comment by lervag** [comment]
Can you reproduce this with a minimal vimrc file?

**Comment by danielfalbo** [comment]
I get the error too with this minimal `init.vim`:
```vim
call plug#begin('~/.vim/autoload/plugged')
Plug 'lervag/vimtex'
call plug#end() = 'nvr'
```

**Comment by lervag** [comment]
Ah, great, thanks. I've reproduced it. Will look into it asap.

**Comment by lervag** [comment]
Can you test if the latest version fixes the issue?

**Comment by oponkork** [comment]
> Can you test if the latest version fixes the issue?

@lervag, I faced the same issue  a while ago, can confirm it goes away with the latest commit.

**Comment by lervag** [comment]
Great, happy to hear it! Let me know if it persists or if you find similar problems!

**Comment by Tongzhoutz** [comment]
Hello @lervag, thanks for your timely update, I believe the error has been diffused. However, I noticed that my designated snippets in math mode (snippets starting with  `context "math()"` )   still cannot be triggered, while other snippets in normal mode work well.

I an not sure whether this issue is owing to vimtex or other plugins? Thanks.

**Comment by danielfalbo** [comment]
Yeah, my `context "math()"` snippets are not working either. 

`tex.snippets` file to recreate: 
```snippets
global !p
texMathZones = ['texMathZone' + x for x in ['A', 'AS', 'B', 'BS', 'C', 'CS', 'D', 'DS', 'E', 'ES', 'F', 'FS', 'G', 'GS', 'H', 'HS', 'I', 'IS', 'J', 'JS', 'K', 'KS', 'L', 'LS', 'DS', 'V', 'W', 'X', 'Y', 'Z', 'AmsA', 'AmsB', 'AmsC', 'AmsD', 'AmsE', 'AmsF', 'AmsG', 'AmsAS', 'AmsBS', 'AmsCS', 'AmsDS', 'AmsES', 'AmsFS', 'AmsGS' ]]

texIgnoreMathZones = ['texMathText']

texMathZoneIds = vim.eval('map('+str(texMathZones)+", 'hlID(v:val)')")
texIgnoreMathZoneIds = vim.eval('map('+str(texIgnoreMathZones)+", 'hlID(v:val)')")

ignore = texIgnoreMathZoneIds[0]

def math():
	synstackids = vim.eval("synstack(line('.'), col('.') - (col('.')>=2 ? 1 : 0))")
	try:
		first = next(i for i in reversed(synstackids) if i in texIgnoreMathZoneIds or i in texMathZoneIds)
		return first != ignore
	except StopIteration:
		return False
endglobal

context "math()"
snippet EE "exists" iA
\exists 
endsnippet
```

`EE` is not automatically triggered even if if has [the `A` option](https://github.com/SirVer/ultisnips/blob/8554371b57c8989cf73f73f288c456fb3f2a3a3a/doc/UltiSnips.txt#L680), and if I force it by pressing my snippets triggering key, this is the result:
```tex
    \exists 
    endsnippet
```
while the expected behaviour is, of course, just
```
    \exists
```

**Comment by oponkork** [comment]
@Tongzhoutz @danielfalbo

It is probably because the math region syntax ids in the new syntax provided by vimtex now begin with `texMathRegion` so the current implementation of the `math()` in ultisnips will not be able to check if you are in math region or not. @lervag will know better.

I will try to see if I can modify it and get it working.

**Comment by lervag** [comment]
> @Tongzhoutz Hello @lervag, thanks for your timely update, I believe the error has been diffused. However, I noticed that my designated snippets in math mode (snippets starting with `context "math()"` ) still cannot be triggered, while other snippets in normal mode work well.

I believe this is more or less the same issue as one I solved earlier today, see [my answer here]( https://github.com/lervag/vimtex/issues/1863#issuecomment-733124368). Let me know if this does not solve your snippet context issue, but if so, please provide more details (i.e. definition of the relevant snippet).

> @danielfalbo Yeah, my `context "math()"` snippets are not working either.

See above.

> @oponkork It is probably because the math region syntax ids in the new syntax provided by vimtex now begin with `texMathRegion` so the current implementation of the `math()` in ultisnips will not be able to check if you are in math region or not. @lervag will know better.

Exactly that.

**Comment by danielfalbo** [comment]
Yeah, this solves the issue for me :) Thank you much

**Comment by oponkork** [comment]
@lervag Thanks for the link, I was just going through the docs to see what should replace
 `texIgnoreMathZones = ['texMathText']`.

This should fix it. :100: :rocket:

**Comment by lervag** [comment]
Happy to hear it works :)

**Comment by lervag** [comment]
Hi all: Please note that the newly released v2.1 reverts the naming scheme `tex{type}Region` back to `tex{type}Zone`. Thus, whereever you refer to `texMathRegion` you should now change it back to `texMathZone`.
